include/ck/tensor_operation/gpu/device/masking_specialization.hpp

```diff
@@ -47,13 +50,39 @@ struct MaskOutUpperTrianglePredicate
         return operator()(m + m_tile - 1, n);
     }
 };
+struct MaskUpperTriangleFromBottomRightPredicate
+{
+    MaskUpperTriangleFromBottomRightPredicate() : offset_(0) {}
+    __host__ __device__ void SetOffset(const index_t offset) { offset_ = offset; }
+    __host__ __device__ constexpr bool operator()(index_t m, index_t n) const
+    {
+        return n > (m + offset_);
+    }
+
+    __host__ __device__ constexpr bool
+    IsTileSkippable(index_t m, index_t n, index_t m_tile, index_t /*n_tile*/) const
+    {
+        return operator()(m + m_tile - 1, n);
+    }
+
+    private:
+    index_t offset_;
+};
 
 // to track the points which need to be set to -inf on C0
 // Note: no need to reset M padding value, because they will not be stored out.
 template <typename MaskOutPredicate>
 struct C0MatrixMask_impl
 {
-    C0MatrixMask_impl(index_t NRaw) : NRaw_(NRaw), predicate_(MaskOutPredicate{}) {}
+    C0MatrixMask_impl(index_t MRaw, index_t NRaw) : NRaw_(NRaw), predicate_(MaskOutPredicate{})
+    {
+        if constexpr(std::is_same<MaskOutPredicate,
+                                  MaskUpperTriangleFromBottomRightPredicate>::value)
+        {
+            if(NRaw > MRaw)
+                predicate_.SetOffset(NRaw - MRaw);
+        }
+    }
```

Comment:
Suggest:  don't call `predicate_.SetOffset()` here, assume the `predicate` is already completely constructed,  just pass and store the predicate object here.  And for  `MaskUpperTriangleFromBottomRightPredicate`, remove `SetOffset`, the constructor is enough, cause it is the user to provide correct diagonal_offset